src/pages/candidate-details/candidate-details.ts

```diff
@@ -77,27 +99,47 @@ export class CandidateDetailsPage extends BasePageComponent implements OnInit, O
       ),
     }
 
-    const { name$, time$, details$ } = this.pageState;
+    const { specialNeeds$, entitlementCheck$, candidateId$, name$, time$, details$ } = this.pageState;
 
     const merged$ = merge(
       name$,
       time$,
       details$.pipe(
         map(details => this.testCategory = details.testCategory.icon as TestCategory)
-      )
+      ),
+      specialNeeds$,
+      entitlementCheck$,
+      candidateId$,
     );
 
     this.subscription = merged$.subscribe();
+    if (this.slotChanged) {
+       this.analytics.logEvent(AnalyticsEventCategories.JOURNAL, AnalyticsEvents.SLOT_CHANGE_VIEWED, this.slotId.toString());
+    }
+    this.store$.dispatch(new ClearChangedSlot(this.slotId));
+  }
+
+  startAnalytics([specialNeeds, entitlementCheck, candidateId]: any[]): void {
+    this.analytics.addCustomDimension(AnalyticsDimensionIndices.DEVICE_ID, this.analytics.uniqueDeviceId);
+    this.analytics.addCustomDimension(AnalyticsDimensionIndices.CANDIDATE_ID, candidateId);
+    this.analytics.addCustomDimension(AnalyticsDimensionIndices.CANDIDATE_WITH_SPECIAL_NEEDS, specialNeeds ? '1' : '0');
+    this.analytics.addCustomDimension(AnalyticsDimensionIndices.CANDIDATE_WITH_CHECK, entitlementCheck ? '1' : '0');    
+    this.analytics.setCurrentPage(AnalyticsScreenNames.CANDIDATE_DETAILS);
   }
 
   ngOnDestroy(): void {
     this.subscription.unsubscribe();
   }
   
   ionViewDidEnter(): void {
-    this.analytics.setCurrentPage(AnalyticsScreenNames.CANDIDATE_DETAILS);
-  }
+    const dimensions$ = zip(
+      this.pageState.specialNeeds$,
+      this.pageState.entitlementCheck$, 
+      this.pageState.candidateId$
+    );
 
+    dimensions$.subscribe(this.startAnalytics.bind(this));
```

Comment:
Are we sure this subscription is cleaned up by the `merge` above? Also, do we get metrics emitted for navigation back to the same candidate as the previous view?